File: note-keeping-frontend/src/services/NotesService.js
import axios from "axios";

const NOTES_BASE_URL = "http://localhost:8080/app/v1";

class NotesService {
  addUser(user) {
    return axios.post(NOTES_BASE_URL + "/user", user);
  }

  getUser(id) {
    return axios.get(NOTES_BASE_URL + "/user/" + id);
  }
  getNotes(uid) {
    return axios.get(NOTES_BASE_URL + "/user/" + uid + "/notes");
  }

  createNote(note, uid) {
    return axios.post(NOTES_BASE_URL + "/user/" + uid + "/notes", note);
  }

  updateNote(note, uid, id) {
    return axios.put(NOTES_BASE_URL + "/user/" + uid + "/notes/" + id, note);
  }

  deleteNote(id, uid) {
    return axios.delete(NOTES_BASE_URL + "/user/" + uid + "/notes/" + id);
  }

  getNoteById(id, uid) {
    return axios.get(NOTES_BASE_URL + "/user/" + uid + "/notes/" + id);
  }

  getNoteByTitle(title, uid) {
    return axios.get(NOTES_BASE_URL + "/user/" + uid + "/notes/" + title);
  }
}

export default new NotesService();

File: note-keeping-backend/src/main/java/com/example/notekeepingbackend/service/NotesService.java
package com.example.notekeepingbackend.service;


import java.util.List;

import com.example.notekeepingbackend.model.Note;
import com.example.notekeepingbackend.model.User;

public interface NotesService {
	
	void addNote(Note note,String id);
	List<Note> listNotes(String uid);
	Note getNote(String uid,long id);
	void deleteNote(String uid,long id);
	void updateNote(Note note,String uid,long id);
	void addUser(String email_id);
	User getUser(String id);
//	Note findNoteByTitle(String uid,String title);

}
